docs/nimble-spright-requirements-adr.md

```diff
@@ -0,0 +1,88 @@
+# ADR: UI Component Contribution Requirements
```

Comment:
What are the next steps once this is approved? We've talked about one or more PRs to create the spright packages in the monorepo. But I want to make sure we don't miss other steps. 

We could brainstorm it in an HLD but I think we might be able to get by with brainstorming a checklist, capturing that in one or more issues, and then working through it. If you agree, could you create the issue(s)?

I'll start with a brain dump here but if you want to separate this conversation from the ADR, we can:
1. Create packages in nimble repo (including integration with all the build/lint/test/publish steps). As Milan was hinting yesterday we might want to split this into a few PRs.
2. Configure CODEOWNERS for the new packages. I'd propose keeping the same owners as each corresponding Nimble package to start with.
3. Create storybook for spright components (I'm still not clear on the architecture here - does this require refactoring any Nimble storybook infrastructure?)
4. Link to spright storybook from relevant places like nimble.ni.dev and Nimble storybook
4. Do we want example apps too? I'd vote yes.
5. Do we want performance tests? I'd be fine to skip this for now.
6. Any changes to issue or PR templates? I can't think of any.
7. Any changes to Renovate? I think it'll work for spright automatically
10. Update repo-wide README / CONTRIBUTING to mention Spright
11. Update spec instructions and templates to consider Spright (e.g. annotate existing templates to say which sections are skippable for Spright contributions)
12. Is #1682 still needed or will those utilties now be accessible because they're in the monorepo?
13. create a demo component in spright and publicize in various forums
8. #1890 
9. Deprecate spright repo and turn off Renovate
15. ask Brandon to create a sweet new logo

Open to feedback on what's missing or not needed from this list!

Reply:
I have created the following issues which should track all this work:
- #1974 
- #1975 
- #1976 
- #1977 
- #1978 
- #1979 

The last is kind of a catch-all for small tasks that don't warrant a separate issue. I'm not quite sure what refactoring will be necessary to share code/configuration between Nimble and Spright. I don't think we need to work that out ahead of time, and we may end up creating tech debt items as needed.